### PR TITLE
Use LinearRgba in `bevy_gizmos` internals

### PR DESCRIPTION
**Pull request by alice-i-cecile**
# Objective

This PR arose as part of the migration process for `bevy_color`: see #12056.

While examining how `bevy_gizmos` stores color types internally, I found that rather than storing a `Color` internally, it actually stores a `[f32;4]` for a linear RGB, type aliased to a `ColorItem`.

While we don't *have* to clean this up to complete the migration, now that we have explicit strong typing for linear color types we should use them rather than replicating this idea throughout the codebase.

## Solution

- Added `LinearRgba::NAN`, for when you want to do cursed rendering things.
- Replaced the internal color representation in `bevy_gizmo`: this was `ColorItem`, but is now `LinearRgba`.
- `LinearRgba` is now `Pod`, enabling us to use the same fast `bytemuck` bit twiddling tricks that we were using before. This requires:

1. Forcing `LinearRgba` to be `repr(C)`
2. Implementing `Zeroable`, and unsafe trait which defines what the struct looks like when all values are zero.
3. Implementing `Pod`, a marker trait with stringent safety requirements that is required for "plain old data".

### CODE REVIEW

**Review by viridia — APPROVED** [review]

**Review by afonsolage — APPROVED** [review]

File: crates/bevy_gizmos/src/gizmos.rs [hunk 1]
```diff
@@ -19,14 +20,13 @@ use crate::{
...
 #[derive(Resource, Default)]
 pub(crate) struct GizmoStorage<T: GizmoConfigGroup> {
-    pub list_positions: Vec<PositionItem>,
-    pub list_colors: Vec<ColorItem>,
-    pub strip_positions: Vec<PositionItem>,
-    pub strip_colors: Vec<ColorItem>,
+    pub(crate) list_positions: Vec<PositionItem>,
+    pub(crate) list_colors: Vec<LinearRgba>,
+    pub(crate) strip_positions: Vec<PositionItem>,
+    pub(crate) strip_colors: Vec<LinearRgba>,
```
Comment (james7132) [inline crates/bevy_gizmos/src/gizmos.rs:29]: As far as I can tell, making these pub(crate) makes this a breaking change.

**Review by james7132 — APPROVED** [review]

**Comment by IceSentry** [comment]
I'm not sure about the Pod/bytemuck trick. I think LinearRgba should implement ShaderType since it's the color intended for shaders.

**Comment by IceSentry** [comment]
I implemented the ShaderType trait for LinearRgba in this PR https://github.com/bevyengine/bevy/pull/12137

File: crates/bevy_gizmos/src/gizmos.rs (hunk 1 above)
Comment (alice-i-cecile) [reply crates/bevy_gizmos/src/gizmos.rs:29]: The parent type was already `pub(crate)`: this is just a clarity tweak, not a functional difference.

**Comment by alice-i-cecile** [comment]
> [6:30 PM]IceSentry: @Alice 🌹 actually, I misunderstood the gizmo code. You can't really use all the encase goodies for vertex buffers. So I think the bytemuck thing is unavoidable here
> [6:30 PM]IceSentry: It's still a good thing to use ShaderType in LinearRgba, but it can't help much in this case
> [6:31 PM]IceSentry: we definitely don't want to allocate in prepare_asset like that 
> [6:31 PM]IceSentry: it will kill perf
> [6:32 PM]IceSentry: I think there's probably a safer api that's missing here, but for now I think we should use the bytemuck trick 
> [6:34 PM]Alice 🌹: Yeah, that was my concern seeing the final diff 😄 I'll revert it and merge then 🙂

[Conversation on Discord](https://discord.com/channels/691052431525675048/743663924229963868/1211817277616554004) about the final reverted commit.

**Review by IceSentry — APPROVED** [review]
Not relevant to this PR, but something similar could be done for PositionItem using a Vec3 instead.

LGTM

File: crates/bevy_gizmos/src/gizmos.rs (hunk 1 above)
Comment (james7132) [reply crates/bevy_gizmos/src/gizmos.rs:29]: Ah OK, missed that. Thanks!

**Pull request merged**